lib/routes/objects.js

```diff
@@ -30,6 +31,13 @@ module.exports = async function (fastify) {
           throw fastify.httpErrors.notFound();
         }
         version = obj.latestVersion;
+      } else {
+        const objVersions = await fastify.getAllObjectVersions({ name, env });
+        if (!objVersions.includes(version)) {
+          throw fastify.httpErrors.badRequest(
```

Comment:
This isn't really a bad request. It's just not a found version. I'd suggest returning a 404 here instead.

Reply:
Yeah I guess so. Reason why I decided to return a BadRequest (instead of a NotFound)  is to highlight the fact that object itself exists, but it's the version that does not. But anyway, probably a notFound is the best way to go here.

Reply:
Will change this tomorrow.